tests/Integration/Behaviour/Features/Scenario/Order/order_cancel_product.feature

```diff
@@ -0,0 +1,58 @@
+# ./vendor/bin/behat -c tests/Integration/Behaviour/behat.yml -s order --tags order-cancel-product
+@reset-database-before-feature
+Feature: Cancel Order Product from Back Office (BO)
+  In order to cancel order products for FO customers
+  As a BO user
+  I need to be able to cancel an order's products from the BO
+
+  Background:
+    Given email sending is disabled
+    And the current currency is "USD"
+    And country "US" is enabled
+    And the module "dummy_payment" is installed
+    And I am logged in as "test@prestashop.com" employee
+    And there is customer "testCustomer" with email "pub@prestashop.com"
+    And customer "testCustomer" has address in "US" country
+    And I create an empty cart "dummy_cart" for customer "testCustomer"
+    And I select "US" address as delivery and invoice address for customer "testCustomer" in cart "dummy_cart"
+    And I add 5 products "Mug The best is yet to come" to the cart "dummy_cart"
+    And I add 3 products "Mug Today is a good day" to the cart "dummy_cart"
+    And I watch the stock of product "Mug The best is yet to come"
+    And I watch the stock of product "Mug Today is a good day"
+
```

Comment:
You should add some scenarii tests related to the order state (of course you need to manage it in the handler as well)
I think cancel product is forbidden has soon as the order has an invoice?
Her's an example https://github.com/PrestaShop/PrestaShop/pull/17416/files#diff-0368d4838ab98bab3f54376d260b7136R554

Reply:
yes, and also checking the canceled status of the order (when everything has been canceled)

Reply:
done, I also checked that order state is set to cancelled after all products have been cancelled:

https://github.com/PrestaShop/PrestaShop/pull/17149/commits/f09de8b66abe7f83d6442d4df2dfb4fd28827bca